common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AbstractAmazonServiceProcessor.java

```diff
@@ -162,6 +163,8 @@ protected void createClientBuilders(
             Function<RuntimeValue<SdkHttpClient.Builder>, RuntimeValue<AwsClientBuilder>> syncClientBuilderFunction,
             Class<?> asyncClientBuilderClass,
             Function<RuntimeValue<SdkAsyncHttpClient.Builder>, RuntimeValue<AwsClientBuilder>> asyncClientBuilderFunction,
+            Class<?> presignerBuilderClass,
+            RuntimeValue<SdkPresigner.Builder> presignerBuilder,
```

Comment:
Is it specific to S3? Because if so, I wonder if it should be done in a separate method rather than this one. That would avoid passing `null` parameters for all the other implementations.

It's just a suggestion, I haven't looked at the code closely.

Reply:
Well AWS has a base `SdkPresigner.Builder` and a derived `S3Presigner.Builder`. I'm actually not sure if there are other presigners. The SDK is certainly setup that way but I never checked for any others.

Reply:
OK, if it follows how the SDK is designed, it's good enough for me.